src/block.ts

```diff
@@ -75,6 +80,7 @@ const parser = (docParser: DocParser) => (block: BlockElement): Result<string, B
                     displayCredit,
                     credit,
                     file,
+                    typeData: masterAsset?.typeData
```

Comment:
`masterAsset` is available as a non-optional in the `asset` argument here, so I think you can do:

```ts
const imageBlock: Option<Result<string, Block>> = fromNullable(masterAsset)
  .map(asset => new Ok({
    kind: ElementType.IMAGE,
    alt,
    caption,
    displayCredit,
    credit,
    file: asset.file,
    width: asset.typeData.width,
    height: asset.typeData.height,
}));
```